fe/fe-core/src/main/java/com/starrocks/sql/optimizer/task/DeriveStatsTask.java

```diff
@@ -44,9 +44,9 @@ public void execute() {
         }
 
         boolean needDerivedChildren = false;
-        // If we haven't got enough stats to compute the current stats, derive them
-        // from the child first
-        for (int i = 0; i < groupExpression.arity(); ++i) {
+        // If we haven't got enough stats to compute the current stats, derive them from the child first.
+        // For CTE, we need derive right tree first, then derive left
```

Comment:
derive right child first？

Reply:
It's work on stack, first input last out, so it's left child first